src/cdk/schematics/utils/build-component.ts

```diff
@@ -40,6 +40,20 @@ import {
 import {getProjectFromWorkspace} from './get-project';
 import {getDefaultComponentOptions} from './schematic-options';
 
+/**
+ * Build a default project path for generating.
+ * @param project The project to build the path for.
+ */
+function buildDefaultPath(project: WorkspaceProject): string {
```

Comment:
Why can't we use `buildDefaultPath` from `utility/workspace` as in https://github.com/angular/angular-cli/blob/master/packages/schematics/angular/component/index.ts#L138?

All of this code is basically a copy of the `component` schematic `@schematics/angular` (with small modifications), but I'd prefer to keep it as similar as possible where possible.

Reply:
Because `buildDefaultPath` from `utility/workspace` requires the usage of the new workspace API.

We can refactor this schematic to use the new workspace API, that was introduced in version 8 and is used in the other schematic. That said, all of those method used there, are from internal API's.

Let me know which one to you prefer.

Reply:
I think we should consider switching, but it's unclear to me how that will affect consumers of the Angular Material schematics if they are on an older version of the CLI.

The code here is probably not really too much in sync with `@schematics/angular/component` anymore, so I'm fine with the current copying. We can consider the new workspace API in a follow-up.